src/frontend/src/lib/utils/results.utils.ts

```diff
@@ -0,0 +1,13 @@
+import type { ResultSuccess, ResultSuccessReduced } from '$lib/types/utils';
+import { nonNullish } from '@dfinity/utils';
+
+export const reduceResults = <T = unknown>(
+	results: [ResultSuccess<T>, ...ResultSuccess<T>[]]
+): ResultSuccessReduced<T> =>
+	results.reduce(
+		(acc, { success: s, err: e }) => ({
+			success: acc.success && s,
+			err: nonNullish(e) ? [...acc.err, e] : acc.err
+		}),
+		{ success: true as boolean, err: [] as T[] }
```

Comment:
- why `as boolean`?
- if you try `results.reduce<ResultSuccess<T>>(acc ...`, do you maybe not need any cast?

Reply:
Thanks. I learned about the second thanks to an eslint rule which is set by default in the `oisy-wallet-signer` project. Maybe we can set the same rule in this repo if you like it too?

Reply:
yes! always in favour of adding eslint rules eheh! will do in a separate PR

Reply:
> just as info, is it "standard-with-typescript" ruleset?

As you feel its best

Reply:
Done with:

- https://github.com/dfinity/oisy-wallet/pull/2343